README.md

```diff
@@ -75,8 +75,8 @@ The error object can be passed along with the original source to `Parsimmon.form
     any character except one of the characters `"a"`, `"b"`, or `"c"`,
     and will yield the same.
   - `Parsimmon.regex(/myregex/, group=0)` is a parser that expects the stream
-    to match the given regex, and yields the given match group, or the
-    entire match.
+    to match the given regexp, and yields the given match group, or the
+    entire match. The regexp may only use the following flags: `imu`. `Parsimmon.regex` will throw an error for any other flag. The regexp is also wrapped in `^(?: regexp )` because it doesn't make sense to match something further away than the next character. Greedy regexp should be wrapped like this: `( regexp )+` (e.g. `Parsimmon.regex(/(a+|a*b)+/)`).
```

Comment:
> The regexp may only use the following flags: `imu`

Instead, this should just say that the regexp must *not* use the `g` or `y` flags.

> The regexp is also wrapped in `^(?: regexp )` because it doesn't make sense to match something further away than the next character.

I don't think this is necessary at all. The process is completely transparent. You can just say "matching begins at the parser's cursor" if you want to say anything at all.

> Greedy regexp should be wrapped like this: ...

I don't think this is valuable. What does "greedy" even mean in this context? It is obvious to anyone with *any* knowledge of regexps that they can use `+` to repeat the match as many times as is possible. We don't need a regexp basics tutorial to document this parser.